runtime/modules/xml/core/src/test/java/io/atlasmap/xml/core/XmlFieldReaderTest.java

```diff
@@ -359,4 +366,673 @@ private String getDocumentString(String uri) throws IOException {
         fis.close();
         return new String(buf);
     }
+
+    @Test
+    public void testXmlFieldDoubleMax() throws Exception {
+        String fieldPath = "/primitive/value";
```

Comment:
Can we make some private/protected methods for repetitious code? There seems to a few standard code patterns in use in the file. Make some methods which parametrize differing values.

Reply:
Done. extracted private method for repetitive codes. Please review. Thanks.